docs/backend-api/api-methods.md

```diff
@@ -176,36 +179,255 @@ You will need to generate new token after the first one expires.
+      }
+    }
+    ```
+
+* **Request Body for specified UUID**
+
+  * **Content**
+    ```
+    {
+      "uuid": "7a34a8cf-8f3b-405d-9662-f20d983cfc8a",
+      "name": {
+          "en": "22\" Sencor SLE 22F46DM4 HELLO KITTY",
+          "cs": "22\" Sencor SLE 22F46DM4 HELLO KITTY"
+      },
+      "hidden": false,
+      "sellingDenied": false,
+      "sellingFrom": "2000-01-16T00:00:00+00:00",
+      "sellingTo": "2100-01-16T00:00:00+00:00",
+      "catnum": "9177759",
+      "ean": "8845781245930",
+      "partno": "SLE 22F46DM4",
+      "shortDescription": {
+          "1": "Television LED, 55 cm diagonal ...",
+          "2": "Sencor SLE 22F46DM4 Hello Kitty je ..."
+      },
+      "longDescription": {
+          "1": "Television LED, 55 cm diagonal ...",
+          "2": "<p><strong>Sencor SLE 22F46DM4 ..."
+      }
+    }
+    ```
+
+* **Success Response**
+
+  * **Code** `201`  
+    **Headers**
+    `Location: http://127.0.0.1:8000/api/v1/products/7a34a8cf-8f3b-405d-9662-f20d983cfc8a`
+
+* **Error Response**
+
+  * **Code** `401 UNAUTHORIZED`
+
+  OR
+
+  * **Code** `415 Unsupported Media Type`
+
+  OR
+
+  * **Code** `400 Bad Request`  
+    **Content**
+    ```
+    {
+        "message": "Provided data did not pass validation",
+        "errors": {
+            "sellingTo": "The value \"12.1.2019\" is not a valid DateTime::ATOM format.",
+            "longDescription.1": "The value 345 is not a valid string."
+        }
+    }
+    ```
+
+    OR
+
+  * **Code** `422 Unprocessable Entity`  
+    **Content**
+    ```
+    {
+        "message": "Product with 74da76db-61ab-43fb-926e-43ba326325fc UUID already exists"
+    }
+    ```
+
+**Partial Product Update**
+----
+
+* **URL**
+
+  `/api/v1/products/{uuid}`
+
+* **Method**
+
+  `PATCH`
+
+* **Headers**
+
+  `Content-Type: application/json`
+
+*  **URL Params**
+
+   **Required**
+
+   `uuid=[string]`
+
+* **Request Body**
+
+  * **Content**
+    ```
+    {
+      "name": {
+          "en": "22\" Sencor SLE 22F46DM4 HELLO KITTY",
+          "cs": "22\" Sencor SLE 22F46DM4 HELLO KITTY"
+      },
+      "hidden": false,
+      "shortDescription": {
+          "1": "Television LED, 55 cm diagonal ..."
+      }
+    }
+    ```
+
+* **Success Response**
+
+  * **Code** `204`
+
+* **Error Response**
+
+  * **Code** `401 UNAUTHORIZED`
+
+  OR
+
+  * **Code** `415 Unsupported Media Type`
+
+  OR
+
+  * **Code** `400 Bad Request`  
+    **Content**
+    ```
+    {
+        "message": "Provided data did not pass validation",
+        "errors": {
+            "sellingTo": "The value \"12.1.2019\" is not a valid DateTime::ATOM format.",
+            "longDescription.1": "The value 345 is not a valid string."
+        }
+    }
+    ```
+
+  OR
+
+  * **Code** `404 Not Found`
+
+*Note: Updating variant/main variant is not supported yet*
+
+**Delete Product**
+----
+
+* **URL**
+
+  `/api/v1/products/{uuid}`
+
+* **Method**
+
+  `DELETE`
+
+*  **URL Params**
```

Comment:
DELETE method also checks for content-type and can return error-response

Reply:
added documentation (I don't want to hack FOSRest bundle to skip this method)